gettsim/pre_processing/policy_for_date.py

```diff
@@ -49,61 +50,146 @@
 
 
 def get_policies_for_date(policy_date, groups="all"):
-    if isinstance(policy_date, str):
-        policy_date = pd.to_datetime(policy_date)
+    """Get the state of the policy system for a particular date.
+
+    The function returns time dependent policy reforms (functions) as well as the
+    policy parameters for the current policy system.
+
+    Parameters
+    ----------
+    policy_date : datetime.date, str, int
+                  The date for which the policy system is set up.
+    groups : list, str
+             The group or a list of groups which parameters are loaded. Default is
+             all parameters
+
+    Returns
+    -------
+    params_dict : dict
+                Dictionary of parameters grouped in policy system compartments given
+                in groups.
+    policy_func_dict : dict
+                Dictionary of time dependent policy reforms. Keys are the
+                variable names they create.
+
+
+    """
+    # Check policy date for correct format and transfer to datetime.date
+    policy_date = check_date(policy_date)
+
+    # Check groups argument for correct format and transfer to list.
+    group_list = check_groups(groups)
+
+    params_dict = {}
+
+    for group in group_list:
+        tax_data = load_data(policy_date, group)
+
+        # Align paramters for e.g. piecewise polynomial functions
+        params_dict[group] = align_parameters(tax_data)
+
+    policy_func_dict = load_reforms_for_date(policy_date)
+
+    return params_dict, policy_func_dict
+
+
+def align_parameters(tax_data):
+    """Check if parameters are stored in implicit structures and align to general
+    structure.
+
+    Parameters
+    ----------
+    tax_data : dict
+            Loaded raw tax data.
+
+    Returns
+    -------
+
+    """
+    for param in tax_data:
+        if type(tax_data[param]) == dict:
```

Comment:
`isinstance` checks are preferred in Python.
```suggestion
        if isinstance(tax_data[param], dict):
```